pkg/selectors/kernel.go

```diff
@@ -363,7 +363,48 @@ func writeMatchValuesInMap(k *KernelSelectorState, values []string, ty uint32) e
 			return fmt.Errorf("Unknown type: %d", ty)
 		}
 		m[val] = struct{}{}
+	}
+	// write the map id into the selector
+	WriteSelectorUint32(k, mid)
+	return nil
+}
 
+func writeMatchRangesInMap(k *KernelSelectorState, values []string, ty uint32) error {
+	mid, m := k.newValueMap()
+	for _, v := range values {
+		var rangeVal [2]uint64
+		rangeStr := strings.Split(v, ":")
+		if len(rangeStr) > 2 {
+			return fmt.Errorf("MatchArgs value %s invalid: range should be 'min:max'", v)
+		} else if len(rangeStr) == 1 {
```

Comment:
It took me a while to parse this code. Can we add a comment like:
// In case a user defines a single value `val`, i.e., not a  range,  we treat it as a range of `val:val`.

Reply:
Yep